packages/salesforce-adapter/src/sfdx_parser/sfdx_dump.ts

```diff
@@ -0,0 +1,166 @@
+} from '@salesforce/source-deploy-retrieve'
+import { SfProject } from '@salesforce/core'
+import { isSubDirectory, rm } from '@salto-io/file'
+import { logger } from '@salto-io/logging'
+import { Adapter } from '@salto-io/adapter-api'
+import { resolveChangeElement } from '@salto-io/adapter-components'
+import { filter } from '@salto-io/adapter-utils'
+import { collections, promises, values } from '@salto-io/lowerdash'
+import { allFilters, NESTED_METADATA_TYPES } from '../adapter'
+import { SYSTEM_FIELDS, UNSUPPORTED_SYSTEM_FIELDS } from '../constants'
+import { getLookUpName } from '../transformers/reference_mapping'
+import { buildFetchProfile } from '../fetch_profile/fetch_profile'
+import SalesforceClient from '../client/client'
+import { createDeployPackage, DeployPackage, PACKAGE } from '../transformers/xml_transformer'
+import { addChangeToPackage, validateChanges } from '../metadata_deploy'
+import { isInstanceOfCustomObjectChangeSync } from '../filters/utils'
+import { SyncZipTreeContainer } from './tree_container'
+
+const log = logger(module)
+const { awu } = collections.asynciterable
+const { withLimitedConcurrency } = promises.array
+
+const FILE_DELETE_CONCURRENCY = 100
+
+const getManifestComponentsToDelete = async (tree: TreeContainer, pkg: DeployPackage): Promise<Set<string>> => {
+  const deletionsFilename = `${PACKAGE}/${pkg.getDeletionsPackageName()}`
+  if (!tree.exists(deletionsFilename)) {
+    return new Set()
+  }
+  const deletionManifest = await new ManifestResolver(tree).resolve(deletionsFilename)
+
+  return new Set(deletionManifest.components.map(comp => `${comp.type.id}.${comp.fullName}`))
+}
+
+const getComponentsToDelete = async (
+  tree: TreeContainer,
+  pkg: DeployPackage,
+  currentComponents: SourceComponent[],
+): Promise<SourceComponent[]> => {
+  const manifestDeletions = await getManifestComponentsToDelete(tree, pkg)
+  const toDelete = currentComponents
+    // source components only returns top level components, if we want to delete child components (e.g - fields)
+    // we have to explicitly iterate over those as well
+    .flatMap(comp => [comp].concat(comp.getChildren()))
+    .filter(comp => manifestDeletions.has(`${comp.type.id}.${comp.fullName}`))
+  // If a component was deleted, we should also delete all of its child components
+  const childComponentsToDelete = toDelete.flatMap(comp => comp.getChildren())
+  return toDelete.concat(childComponentsToDelete)
+}
+
+const compactPathList = (paths: string[]): string[] => {
+  // Remove paths that are nested under other paths to delete so we don't try to double-delete
+  let lastPath: string | undefined
+  return paths
+    .sort()
+    .map(pathToDelete => {
+      if (lastPath !== undefined && isSubDirectory(pathToDelete, lastPath)) {
+        return undefined
+      }
+      lastPath = pathToDelete
+      return pathToDelete
+    })
+    .filter(values.isDefined)
+}
+
+type DumpElementsToFolderFunc = Required<Adapter>['dumpElementsToFolder']
```

Comment:
Nit: I think that this is more clear:
```suggestion
type DumpElementsToFolderFunc = NonNullable<Adapter['dumpElementsToFolder']>
```
(mainly because all of `Adapter` doesn't need to be required, only this type)